docs/upgrade-guide.md

```diff
@@ -2,46 +2,55 @@
 
+### Dependencies
+
+deck.gl 4.1 requires luma.gl as peer dependency, but 4.2 specifies it as a normal "dependency". This means that many applications no longer need to list luma.gl in their package.json. Applications that do might get multiple copies of luma.gl installed, which will not work. **luma.gl will detect this situtation during run-time throwing an exception**, but **npm and yarn will not detect it during install time**. Thus your build can look successful but will fail during upgrade.
+
 ### Layer Props
 
-Coordinate system related props have been renamed for clarity.
+Coordinate system related props have been renamed for clarity. The old props are still available but will generate a deprecation warning.
 
 | Layer            | Old Prop           | New Prop             | Comment |
 | ---              | ---                | ---                  | ---     |
 | Layer            | `projectionMode`   | `coordinateSystem`   | Any constant from `COORDINATE_SYSTEM`  |
 | Layer            | `projectionOrigin` | `coordinateOrigin`   | |
 
+Note; There is also an important semantical change in that using `coordinateSystem` instead of `projectionMode` causes the superimposed `METER_OFFSET` system's y-axis to point north instead of south. This was always the intention so in some sense this was regarded as a bug fix.
 
 ### DeckGL component
 
-Following methods and props have been renamed for clarity.
-
+Following methods and props have been renamed for clarity. The semantics are unchanged. The old props are still available but will generate a deprecation warning.
 
 | Old Method            | New Method        | Comment |
 | ---                   | ---               | ---     |
-| `queryVisibleObjects` | `pickObjects`     | |
-| `queryObject`         | `pickObject`      | |
+| `queryObject`         | `pickObject`      | These names were previously aligned with react-map-gl, but ended up confusing users. Since rest of the deck.gl documentation talks extensively about "picking" it made sense to stay with that terminology. |
+| `queryVisibleObjects` | `pickObjects`     | The word "visible" was intended to remind the user that this function only selects the objects that are actually visible in at least one pixel, but again it confused more than it helped. |
 
 
 | Old Prop              | New Props         | Comment |
 | ---                   | ---               | ---     |
-| `useDevicePixelRatio` | `useDevicePixels` | |
+| `useDevicePixelRatio` | `useDevicePixels` | Changed due to feedback that the original name was confusing. |
 
 
-### Picking uniforms
+### Picking Uniforms
 
-`renderPickingBuffer` and `selectedPickingColor` are deprecated, and will be removed in next major version. It is recommended to use `luma.gl` picking module.
+The shader uniforms `renderPickingBuffer` and `selectedPickingColor` are deprecated, and will be removed in next major version. The props are still available but will generate a deprecation warning.
```

Comment:
Actually these uniforms are always set and we don't generate deprecation warnings. We should ideally add deprecation warning if shader contains any of these uniforms, but we are not doing that now.

Reply:
> Actually these uniforms are always set and we don't generate deprecation warnings. 

Good catch. Updated.